spring-grpc-spring-boot-autoconfigure/src/main/java/org/springframework/grpc/autoconfigure/server/GrpcServerMetricAutoConfiguration.java

```diff
@@ -0,0 +1,29 @@
+package org.springframework.grpc.autoconfigure.server;
+
+import io.grpc.ServerBuilder;
+import io.grpc.ServerInterceptor;
+import io.micrometer.core.instrument.binder.grpc.ObservationGrpcServerInterceptor;
+import io.micrometer.observation.ObservationRegistry;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.grpc.server.ServerBuilderCustomizer;
+
+@AutoConfiguration
+@ConditionalOnClass(ObservationGrpcServerInterceptor.class)
+@ConditionalOnBean(ObservationRegistry.class)
+public class GrpcServerMetricAutoConfiguration {
+
+	@Bean
+	public ServerInterceptor observationGrpcServerInterceptor(ObservationRegistry observationRegistry) {
+		return new ObservationGrpcServerInterceptor(observationRegistry);
+	}
+
+	@Bean
+	<T extends ServerBuilder<T>> ServerBuilderCustomizer<T> metricsInterceptor(
+			ServerInterceptor observationGrpcServerInterceptor) {
+		return (serverBuilder) -> serverBuilder.intercept(observationGrpcServerInterceptor);
```

Comment:
This definition has broader effects than the purpose of this PR, since it adds arbitrary `ServerInterceptor`s to the `ServerBuilder`, which should be defined in a separate class.

Reply:
Hi @making , This is a temporary means to apply the interceptor to the builder via auto-configuration. 
This will be replaced by marking the interceptor bean above w/ `@GrpcGlobalServerInterceptor`. I am working on that now.

Reply:
Thanks for the clarification!